### PR TITLE
Implement multiprocessing in file upload

### PR DESCRIPTION
**Pull request by marcorosa**
Use a Pool for uploading shards, instead of launching single threads every time (consistency with the downloader).
For uploading a file, use the method `file_upload` instead of `createNewUploadThread` (more user friendly).
Some class variables have been deleted (e.g., the bucket id.. keeping it as a class variable has no sense)

### CODE REVIEW

**Comment by steenzout** [comment]
only 1 comment.

rest looks good.

**Comment by Miskerest** [comment]
Builds failing on Travis, is this an issue?

**Comment by steenzout** [comment]
@Miskerest the build needs to pass so whatever is making it fail needs to be changed.

**Comment by Miskerest** [comment]
Building passing. Merging & closing.

File: storj/uploader.py [hunk 1]
```diff
@@ -396,13 +379,10 @@ def file_upload(self):
 
         success = False
         try:
-            # TODO
-
-            # This is the actual upload_file method
+            # Post an upload_file request
             response = self.client._request(
```
Comment (marcorosa) [inline storj/uploader.py]: yes, what i wanted to do is providing a working uploader, then we can edit and improve it, if there is some duplicated code

Comment (steenzout) [inline storj/uploader.py]: ok, I added #115 .

**Pull request merged**